[SPARK-4123][Project Infra]: Show new dependencies added in pull requests

brennonyork opened the pull request:
Starting work on this, but need to find a way to ensure that, after doing a checkout from `apache/master`, we can successfully return to the current checkout. I believe that `git rev-parse HEAD` will get me what I want, but pushing this PR up to test what the Jenkins boxes are seeing.

Commit:
beginning work on pr test to add new dependencies

Commit:
removed jenkins tests to grab some values from the jenkins box

SparkQA commented:
[Test build #28886 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28886/consoleFull) for   PR 5093 at commit [`ae83b90`](https://github.com/apache/spark/commit/ae83b90f839d3a9e239b88ee2fcc354ceb4f66c2).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28886 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28886/consoleFull) for   PR 5093 at commit [`ae83b90`](https://github.com/apache/spark/commit/ae83b90f839d3a9e239b88ee2fcc354ceb4f66c2).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28886/
Test FAILed.

Commit:
added a git checkout 'git rev-parse HEAD' to the end of each pr test

SparkQA commented:
[Test build #28887 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28887/consoleFull) for   PR 5093 at commit [`2bca3c3`](https://github.com/apache/spark/commit/2bca3c31bff45e098ad1eb7982cf19bcad64d437).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28887 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28887/consoleFull) for   PR 5093 at commit [`2bca3c3`](https://github.com/apache/spark/commit/2bca3c31bff45e098ad1eb7982cf19bcad64d437).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28887/
Test FAILed.

nchammas commented on dev/tests/pr_new_dependencies.sh:
```diff
@@ -0,0 +1,36 @@
+sha1="$2"
+
+./build/mvn compile dependency:build-classpath | grep -A 5 "Building Spark Project Assembly" | tail -n 1 | tr ":" "\n" | rev | cut -d "/" -f 1 | rev | sort > my-classpath.txt
+git checkout apache/master
```
Do we need to do any kind of cleaning between the two Maven operations?

brennonyork replied:
Honestly I'm unsure. I believe we're going to need to run a `mvn clean compile dependency:build-classpath` for each to work properly, but @pwendell was running it without the `compile` step which I couldn't reproduce locally without an `install` as well. Going to test now and see what Jenkins tells me.

Commit:
adding the new dependencies script into the test mix

SparkQA commented:
[Test build #28888 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28888/consoleFull) for   PR 5093 at commit [`94d3547`](https://github.com/apache/spark/commit/94d35478c8205386ac4ff0e265a0bfbb073bc8c7).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28888 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28888/consoleFull) for   PR 5093 at commit [`94d3547`](https://github.com/apache/spark/commit/94d35478c8205386ac4ff0e265a0bfbb073bc8c7).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28888/
Test FAILed.

Commit:
Capturing output directly from dependency builds

SparkQA commented:
[Test build #28890 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28890/consoleFull) for   PR 5093 at commit [`dae7ba8`](https://github.com/apache/spark/commit/dae7ba8bac01d58a838df8c5c1f904d19a2d8e87).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28890/
Test FAILed.

Commit:
updated new dependencies test

SparkQA commented:
[Test build #28894 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28894/consoleFull) for   PR 5093 at commit [`63a35c9`](https://github.com/apache/spark/commit/63a35c908598074bb0acb2e310a4905fe28502a0).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28894/
Test FAILed.

Commit:
minor updates and style changes

SparkQA commented:
[Test build #28895 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28895/consoleFull) for   PR 5093 at commit [`f8011d8`](https://github.com/apache/spark/commit/f8011d8886e0a2a2db74ae5715cb324eb30eedbb).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28895/
Test FAILed.

Commit:
removing new deps test to isolate why jenkins isn't posting messages

SparkQA commented:
[Test build #28896 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28896/consoleFull) for   PR 5093 at commit [`126ce61`](https://github.com/apache/spark/commit/126ce61580d805b464f7a4534d0be05411ff0e4b).
 * This patch merges cleanly.

SparkQA commented:
[Test build #28896 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/28896/consoleFull) for   PR 5093 at commit [`126ce61`](https://github.com/apache/spark/commit/126ce61580d805b464f7a4534d0be05411ff0e4b).
 * This patch **passes all tests**.


 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/28896/
Test FAILed.

pwendell commented on dev/tests/pr_new_dependencies.sh:
```diff
@@ -0,0 +1,77 @@
+else
+  # Pretty print the new dependencies
+  new_deps=$(echo ${DIFF_RESULTS} | grep "<" | cut -d" " -f2 | awk '{print "   * "$1}')
+  echo " * This patch **adds the following new dependencies:**\n${new_deps}"
```
It would actually be good to show additions and removals of dependencies (in some cases a library will get a new version, which ends up here as an addition and removal). I updated the JIRA title to reflect that.

brennonyork replied:
Was thinking the same thing actually. I'll make sure to include that before this WIP is completed. Thanks!

Commit:
updated location of maven binary

SparkQA commented:
[Test build #29006 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29006/consoleFull) for   PR 5093 at commit [`291a8fe`](https://github.com/apache/spark/commit/291a8fea27d1aadf7db28936ef56762e5d74eb7b).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29006 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29006/consoleFull) for   PR 5093 at commit [`291a8fe`](https://github.com/apache/spark/commit/291a8fea27d1aadf7db28936ef56762e5d74eb7b).
 * This patch **passes all tests**.


 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29006/
Test FAILed.

Commit:
added the removed deps and echo output for jenkins testing

SparkQA commented:
[Test build #29007 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29007/consoleFull) for   PR 5093 at commit [`b98f78c`](https://github.com/apache/spark/commit/b98f78c8f652b27e66d3fe554b9b972927017658).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29007/
Test FAILed.

Commit:
removed the error to dev/null

Commit:
added this_mssg echo output

SparkQA commented:
[Test build #29008 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29008/consoleFull) for   PR 5093 at commit [`6912584`](https://github.com/apache/spark/commit/69125849f4ca32d17e1db6fa47f61c9b992a9a94).

SparkQA commented:
[Test build #29008 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29008/consoleFull) for   PR 5093 at commit [`6912584`](https://github.com/apache/spark/commit/69125849f4ca32d17e1db6fa47f61c9b992a9a94).
 * This patch **passes all tests**.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29008/
Test FAILed.

Commit:
removed the git checkout

SparkQA commented:
[Test build #29014 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29014/consoleFull) for   PR 5093 at commit [`f2abc8c`](https://github.com/apache/spark/commit/f2abc8c49490970f7b0bd5829a0696655beb4c09).

SparkQA commented:
[Test build #29014 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29014/consoleFull) for   PR 5093 at commit [`f2abc8c`](https://github.com/apache/spark/commit/f2abc8c49490970f7b0bd5829a0696655beb4c09).
 * This patch **passes all tests**.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29014/
Test FAILed.

Commit:
updated the unop for ensuring a test is available

SparkQA commented:
[Test build #29017 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29017/consoleFull) for   PR 5093 at commit [`56f74a8`](https://github.com/apache/spark/commit/56f74a8a1e7f4c808827ba1f0b09f1f3b40db028).

SparkQA commented:
[Test build #29017 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29017/consoleFull) for   PR 5093 at commit [`56f74a8`](https://github.com/apache/spark/commit/56f74a8a1e7f4c808827ba1f0b09f1f3b40db028).
 * This patch **passes all tests**.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29017/
Test FAILed.

Commit:
Merge remote-tracking branch 'upstream/master' into SPARK-4123

Commit:
commenting out new deps script

SparkQA commented:
[Test build #29019 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29019/consoleFull) for   PR 5093 at commit [`97e5cfb`](https://github.com/apache/spark/commit/97e5cfb08d91053a46e3c306b8af9593a5fde942).

SparkQA commented:
[Test build #29019 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29019/consoleFull) for   PR 5093 at commit [`97e5cfb`](https://github.com/apache/spark/commit/97e5cfb08d91053a46e3c306b8af9593a5fde942).
 * This patch **passes all tests**.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29019/
Test FAILed.

Commit:
removed escapes for quotes

SparkQA commented:
[Test build #29020 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29020/consoleFull) for   PR 5093 at commit [`e9e3ad1`](https://github.com/apache/spark/commit/e9e3ad1d36eb990e6a9feff3092fafe5f3138389).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29020 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29020/consoleFull) for   PR 5093 at commit [`e9e3ad1`](https://github.com/apache/spark/commit/e9e3ad1d36eb990e6a9feff3092fafe5f3138389).
 * This patch **passes all tests**.


 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29020/
Test FAILed.

Commit:
added back the new deps test

SparkQA commented:
[Test build #29021 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29021/consoleFull) for   PR 5093 at commit [`bad9a3a`](https://github.com/apache/spark/commit/bad9a3a1c63019f9cc9086b7b669b313c0b8008b).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29021 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29021/consoleFull) for   PR 5093 at commit [`bad9a3a`](https://github.com/apache/spark/commit/bad9a3a1c63019f9cc9086b7b669b313c0b8008b).
 * This patch **passes all tests**.


 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29021/
Test FAILed.

Commit:
removed echo

Commit:
updated test output newlines

SparkQA commented:
[Test build #29022 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29022/consoleFull) for   PR 5093 at commit [`6e2890d`](https://github.com/apache/spark/commit/6e2890d8a90d24c3196285dff3c166899687f2a2).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29022/
Test FAILed.

Commit:
proper piping of unnecessary stderr and stdout

SparkQA commented:
[Test build #29025 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29025/consoleFull) for   PR 5093 at commit [`d027f8f`](https://github.com/apache/spark/commit/d027f8f6064cfd2dbaac0ff92e7de0ca9e438ac3).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29025 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29025/consoleFull) for   PR 5093 at commit [`d027f8f`](https://github.com/apache/spark/commit/d027f8f6064cfd2dbaac0ff92e7de0ca9e438ac3).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29025/
Test FAILed.

Commit:
added echo statement so jenkins logs which pr tests are running

SparkQA commented:
[Test build #29028 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29028/consoleFull) for   PR 5093 at commit [`2bb5527`](https://github.com/apache/spark/commit/2bb5527e2dc67dae1b4834eac3aaac07f3a76b32).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29028 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29028/consoleFull) for   PR 5093 at commit [`2bb5527`](https://github.com/apache/spark/commit/2bb5527e2dc67dae1b4834eac3aaac07f3a76b32).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29028/
Test FAILed.

srowen commented on dev/run-tests-jenkins:
```diff
@@ -176,7 +183,8 @@ done
 {
-  timeout "${TESTS_TIMEOUT}" ./dev/run-tests
+  # timeout "${TESTS_TIMEOUT}" ./dev/run-tests
+  echo "hello!"
```
(I know it's for WIP testing, just flagging this to remove it later)

srowen commented on dev/tests/pr_new_dependencies.sh:
```diff
@@ -0,0 +1,85 @@
+CURR_CP_FILE="my-classpath.txt"
+MASTER_CP_FILE="master-classpath.txt"
+
+${MVN_BIN} clean compile dependency:build-classpath 2>/dev/null | \
```
`clean compile` is gonna be slow; does `build-classpath` require it?

brennonyork replied:
Yeah, its required :/ I've tested without it and it fails at building `spark-networking`. This adds on, for each run (of which there are two) around 4.5 mins, so 9mins added to the build time. I also looked at seeing what `sbt` could output, but couldn't find anything. Further thought about this as a special case test and to grab the output from the generic build of spark that happens for each PR, but with having to build against the `master` branch as well that didn't seem like a much better option.

pwendell replied:
If that's the case, we should probably gate this entire thing in a check as to whether any pom.xml files are modified. Then for most builds this will not add any time, since most builds do not modify dependencies.

brennonyork replied:
Sounds like a plan. Once I get this working in a state I like I'll set a gate to check all pom.xml files for changes and, if any show changes, will go ahead and execute the code.

Commit:
Merge remote-tracking branch 'upstream/master' into SPARK-4123

Commit:
removed excess git checkout print statements

SparkQA commented:
[Test build #29168 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29168/consoleFull) for   PR 5093 at commit [`6b3042b`](https://github.com/apache/spark/commit/6b3042b7d2f3b77b2b71dc69eeaed05eebc29571).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29168 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29168/consoleFull) for   PR 5093 at commit [`6b3042b`](https://github.com/apache/spark/commit/6b3042b7d2f3b77b2b71dc69eeaed05eebc29571).
 * This patch **passes all tests**.

 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29168/
Test FAILed.

Commit:
added set blocks to catch any non-zero exit codes and updated output

SparkQA commented:
[Test build #29169 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29169/consoleFull) for   PR 5093 at commit [`9e87208`](https://github.com/apache/spark/commit/9e87208bcee8f8d834bb1db4261f304ef65bc9f6).

SparkQA commented:
[Test build #29169 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29169/consoleFull) for   PR 5093 at commit [`9e87208`](https://github.com/apache/spark/commit/9e87208bcee8f8d834bb1db4261f304ef65bc9f6).
 * This patch **passes all tests**.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29169/
Test FAILed.

Commit:
added -x flag with bash to get insight into what is executing and what isnt

SparkQA commented:
[Test build #29171 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29171/consoleFull) for   PR 5093 at commit [`87f9bea`](https://github.com/apache/spark/commit/87f9bea21a161aa34dff00c9409b00cad71a586e).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29171 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29171/consoleFull) for   PR 5093 at commit [`87f9bea`](https://github.com/apache/spark/commit/87f9bea21a161aa34dff00c9409b00cad71a586e).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch adds no new dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29171/
Test FAILed.

Commit:
adding simple mvn statement to see if command executes and prints compile output

SparkQA commented:
[Test build #29174 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29174/consoleFull) for   PR 5093 at commit [`04747e4`](https://github.com/apache/spark/commit/04747e40629edae542a9e2f88da4601a87720fa4).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29174/
Test FAILed.

Commit:
updated dependency output message and changed compile to package given the jenkins failure output

SparkQA commented:
[Test build #29176 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29176/consoleFull) for   PR 5093 at commit [`62ffd65`](https://github.com/apache/spark/commit/62ffd659a3fb2bb7a3e7bc78e1d49a39ced3f257).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29176 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29176/consoleFull) for   PR 5093 at commit [`62ffd65`](https://github.com/apache/spark/commit/62ffd659a3fb2bb7a3e7bc78e1d49a39ced3f257).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29176/
Test FAILed.

Commit:
testing mvn with package

SparkQA commented:
[Test build #29177 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29177/consoleFull) for   PR 5093 at commit [`f70d8cd`](https://github.com/apache/spark/commit/f70d8cd0c9cd2c00767fd5e2567d7b199fff0180).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29177/
Test FAILed.

Commit:
dont let mvn execute tests

SparkQA commented:
[Test build #29180 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29180/consoleFull) for   PR 5093 at commit [`ea170de`](https://github.com/apache/spark/commit/ea170de4b2961b3e161206209952c36bee8ddb6d).
 * This patch merges cleanly.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29180/
Test FAILed.

Commit:
full test of new deps script

SparkQA commented:
[Test build #29184 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29184/consoleFull) for   PR 5093 at commit [`a18db71`](https://github.com/apache/spark/commit/a18db719705d23e830d045fc723fa5219813eea3).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29184 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29184/consoleFull) for   PR 5093 at commit [`a18db71`](https://github.com/apache/spark/commit/a18db719705d23e830d045fc723fa5219813eea3).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29184/
Test FAILed.

Commit:
added a pom gate check

SparkQA commented:
[Test build #29207 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29207/consoleFull) for   PR 5093 at commit [`2bc7b27`](https://github.com/apache/spark/commit/2bc7b2744a07ec63556ab22850e145d955cd91c5).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29207 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29207/consoleFull) for   PR 5093 at commit [`2bc7b27`](https://github.com/apache/spark/commit/2bc7b2744a07ec63556ab22850e145d955cd91c5).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29207/
Test FAILed.

Commit:
updated to reflect the correct pushed in HEAD variable

SparkQA commented:
[Test build #29208 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29208/consoleFull) for   PR 5093 at commit [`8f3c4b4`](https://github.com/apache/spark/commit/8f3c4b43b098768d89bc970d77d8ffd9441e20a9).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29208 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29208/consoleFull) for   PR 5093 at commit [`8f3c4b4`](https://github.com/apache/spark/commit/8f3c4b43b098768d89bc970d77d8ffd9441e20a9).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29208/
Test FAILed.

Commit:
ensure both gate files exist

SparkQA commented:
[Test build #29209 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29209/consoleFull) for   PR 5093 at commit [`9bce980`](https://github.com/apache/spark/commit/9bce980ab7e24589b67a42114590c3207c825c33).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29209 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29209/consoleFull) for   PR 5093 at commit [`9bce980`](https://github.com/apache/spark/commit/9bce980ab7e24589b67a42114590c3207c825c33).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29209/
Test FAILed.

Commit:
git output to dev null

SparkQA commented:
[Test build #29211 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29211/consoleFull) for   PR 5093 at commit [`5375ad8`](https://github.com/apache/spark/commit/5375ad898eed23e3a5ff7bc346e8606d0115446b).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29211 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29211/consoleFull) for   PR 5093 at commit [`5375ad8`](https://github.com/apache/spark/commit/5375ad898eed23e3a5ff7bc346e8606d0115446b).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29211/
Test FAILed.

Commit:
changed the assembly pom to test whether the pr test will pick up new deps

SparkQA commented:
[Test build #29241 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29241/consoleFull) for   PR 5093 at commit [`27bb9b5`](https://github.com/apache/spark/commit/27bb9b58dabed79b66377279ea74c7adba611bff).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29241 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29241/consoleFull) for   PR 5093 at commit [`27bb9b5`](https://github.com/apache/spark/commit/27bb9b58dabed79b66377279ea74c7adba611bff).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29241/
Test FAILed.

Commit:
updated to print the new dependency found

SparkQA commented:
[Test build #29247 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29247/consoleFull) for   PR 5093 at commit [`e229802`](https://github.com/apache/spark/commit/e229802e0a3cd2e81c21a319b8c3d71c5c8298d7).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29247 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29247/consoleFull) for   PR 5093 at commit [`e229802`](https://github.com/apache/spark/commit/e229802e0a3cd2e81c21a319b8c3d71c5c8298d7).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch **adds the following new dependencies:**
   * <

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29247/
Test FAILed.

Commit:
fixed minor echo issue with quotes

SparkQA commented:
[Test build #29248 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29248/consoleFull) for   PR 5093 at commit [`0074566`](https://github.com/apache/spark/commit/0074566f1724a52f7a57f78709b98846c5f4083f).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29248 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29248/consoleFull) for   PR 5093 at commit [`0074566`](https://github.com/apache/spark/commit/0074566f1724a52f7a57f78709b98846c5f4083f).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch **adds the following new dependencies:**
   * clojure-1.6.0.jar

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29248/
Test FAILed.

Commit:
set new dendencies into a <code> block, removed the bash debugging flag

SparkQA commented:
[Test build #29250 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29250/consoleFull) for   PR 5093 at commit [`7aa1dee`](https://github.com/apache/spark/commit/7aa1dee6ff9a639d9680cd9ac1ef52435e97e3b3).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29250 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29250/consoleFull) for   PR 5093 at commit [`7aa1dee`](https://github.com/apache/spark/commit/7aa1dee6ff9a639d9680cd9ac1ef52435e97e3b3).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch **adds the following new dependencies:**
   * `clojure-1.6.0.jar`

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29250/
Test FAILed.

Commit:
reverted assembly pom, put the regular test suite back in play

SparkQA commented:
[Test build #29251 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29251/consoleFull) for   PR 5093 at commit [`dadd941`](https://github.com/apache/spark/commit/dadd94183ecab3ce1057deec398b475a303ec22b).
 * This patch merges cleanly.

SparkQA commented:
[Test build #29251 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29251/consoleFull) for   PR 5093 at commit [`dadd941`](https://github.com/apache/spark/commit/dadd94183ecab3ce1057deec398b475a303ec22b).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch does not change any dependencies.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29251/
Test PASSed.

brennonyork commented:
/cc @pwendell @srowen @nchammas 

All complete. You can check out build 29250 up a few to get what the output would be like if a new dependency were added. One issue which I'd love to get some opinion on...

Right now the initial post message to Github (i.e. the "Test build started + patch merges cleanly") will take up to 20 minutes to post **if** any `pom.xml` files were changed because it will then run and build both the current PR plus the master branch. This is purely because the "This patch merges cleanly" output is from a `pr_test` and runs in the core test loop. The easiest option to reflect the original way things have been posted would be to move the `pr_merge_ability` test out of the main test loop and have it execute independently. The other option would be to merely post the "Test build started at ..." message and remove the "merges cleanly" portion to be posted with in the post-test message.

I'll admit I'm more in favor of the latter option as I think it keeps things clean as well as the fact that "merges cleanly" is slightly ambiguous given that Github reports on this as well. Thoughts? Whichever way we go I can get that final change up and then I'd say this is ready for review into master.

nchammas commented:
I have advocated in the past for removing the redundant "merges cleanly" message, but IIRC some people found it useful and wanted to keep it.

One way or the other, yes, the mergability test needs to run only once. And more generally, we want the "tests started" message to post as quickly as possible so that contributors know the CI is working.

If that means not posting dependency changes until all tests are done, I personally think that's fair.

srowen commented on dev/tests/pr_new_dependencies.sh:
```diff
@@ -0,0 +1,115 @@
+#+ known as `ghprbActualCommit` in `run-tests-jenkins`
+# Arg2: The SHA1 hash
+#+ known as `sha1` in `run-tests-jenkins`
+#
```
Pretty trivial but the third arg could be described here. I assume the other two scripts simply ignore the third arg.

srowen commented on dev/tests/pr_new_dependencies.sh:
```diff
@@ -0,0 +1,115 @@
+sha1="$2"
+current_pr_head="$3"
+
+MVN_BIN="`pwd`/build/mvn"
```
Is this equivalent to simply `build/mvn`? I see it that way in other scripts.

srowen commented on dev/tests/pr_new_dependencies.sh:
```diff
@@ -0,0 +1,115 @@
+if [ -z "${difference_q}" ]; then
+  echo " * This patch does not change any dependencies."
+else
+# Else we need to manually build spark to determine what, if any, dependencies
```
Nit: indent the comments since they're sort of part of the else block?

srowen commented:
I personally favor a "tests have started" message quickly, yes, but I don't find it essential to print the merge status in that initial message, no. All of those results can be printed in the second "tests have finished" message IMHO. So I think I agree with you.

Commit:
moved starting test output to before pr tests, fixed indentation, changed mvn call to build/mvn

brennonyork commented:
Thanks for the update guys. Per the consensus I moved the "tests have started" message to before the PR tests run. Also, @srowen, updated all items per your comments. Any additional thoughts all?

SparkQA commented:
[Test build #29314 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29314/consoleFull) for   PR 5093 at commit [`42e243e`](https://github.com/apache/spark/commit/42e243e10a11ee7cf4bda38ab12811d88d1ca4d0).

SparkQA commented:
[Test build #29314 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/29314/consoleFull) for   PR 5093 at commit [`42e243e`](https://github.com/apache/spark/commit/42e243e10a11ee7cf4bda38ab12811d88d1ca4d0).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.
 * This patch **adds the following new dependencies:**
   * `avro-1.7.7.jar`
 * This patch **removes the following dependencies:**
   * `avro-1.7.6.jar`

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/29314/
Test PASSed.

srowen commented:
That's looking good to me. It seems to work -- here we actually see the small possibility of a false positive. A separate change updated Avro to 1.7.7 at about the same time that this test ran. I don't think it's a big deal since we can easily reason about it or retest if in doubt.

I'll wait a bit for comments but I like it.

JoshRosen commented:
I've taken this over at #10461, where I found a clean way to speed up the checks to a point where they run near instantaneously without having to modify or delete source files.
